### PR TITLE
[DOC] Change documentation of the default value of ysql_num_shards_per_tserver

### PR DESCRIPTION
**Pull request by sanketkedia**
References GH Issue#[7666](https://github.com/yugabyte/yugabyte-db/issues/7666). There was a code change recently to change the default value of `ysql_num_shards_per_tserver`. Documentation should be updated to reflect this.

### CODE REVIEW

File: docs/content/latest/reference/configuration/yb-master.md [hunk 1]
```diff
@@ -437,7 +437,7 @@ Default: `3`
 
 The number of shards per YB-TServer for each YCQL table when a user table is created.
 
-Default: `-1` (server internally sets default value). For servers with two or less CPU cores, then the default value is `4`. For four or more CPU cores, the default value is `8`. Local cluster installations, created using `yb-ctl` and `yb-docker-ctl`, use a value of `2` for this flag.
+Default: `-1` (server internally sets default value). For servers with two or less CPU cores, then the default value is `4`. For three or more CPU cores, the default value is `8`. Local cluster installations, created with `yb-ctl` and `yb-docker-ctl`, use a value of `2` for this flag. Clusters created with `yugabyted` use a default value of 1.
```
Comment (polarweasel) [inline docs/content/latest/reference/configuration/yb-master.md:440]:
```suggestion
Default: `-1` (server internally sets default value). For servers with up to two CPU cores, the default value is `4`. For three or more CPU cores, the default value is `8`. Local cluster installations, created with `yb-ctl` and `yb-docker-ctl`, use a value of `2` for this flag. Clusters created with `yugabyted` use a default value of `1`.
```

File: docs/content/latest/reference/configuration/yb-master.md [hunk 2]
```diff
@@ -455,7 +455,7 @@ On a per-table basis, the [`CREATE TABLE ... WITH TABLETS = <num>`](../../../api
 
 The number of shards per YB-TServer for each YSQL table when a user table is created.
 
-Default: `8`
+Default: `-1` (server internally sets default value). For servers with two or less CPU cores, then the default value is `2`. For servers with three or four CPU cores, the default value is 4. Beyond 4 cores, the default value is `8`. Local cluster installations, created with `yb-ctl` and `yb-docker-ctl`, use a value of `2` for this flag. Clusters created with `yugabyted` use a default value of 1.
```
Comment (polarweasel) [inline docs/content/latest/reference/configuration/yb-master.md:458]: Also, is the difference between this and YCQL intentional? YCQL jumps straight to `8` for 3+ cores.

```suggestion
Default: `-1` (server internally sets default value). For servers with up to two CPU cores, the default value is `2`. For servers with three or four CPU cores, the default value is `4`. Beyond four cores, the default value is `8`. Local cluster installations, created with `yb-ctl` and `yb-docker-ctl`, use a value of `2` for this flag. Clusters created with `yugabyted` use a default value of `1`.
```

File: docs/content/latest/reference/configuration/yb-tserver.md [hunk 3]
```diff
@@ -329,7 +329,7 @@ Default: `64`
 
 The number of shards per YB-TServer for each YCQL table when a user table is created.
 
-Default: `-1` (server internally sets default value). For servers with two or less CPU cores, then the default value is `4`. For four or more CPU cores, the default value is `8`. Local cluster installations, created with `yb-ctl` and `yb-docker-ctl`, use a value of `2` for this flag.
+Default: `-1` (server internally sets default value). For servers with two or less CPU cores, then the default value is `4`. For three or more CPU cores, the default value is `8`. Local cluster installations, created with `yb-ctl` and `yb-docker-ctl`, use a value of `2` for this flag. Clusters created with `yugabyted` use a default value of 1.
```
Comment (polarweasel) [inline docs/content/latest/reference/configuration/yb-tserver.md:332]:
```suggestion
Default: `-1` (server internally sets default value). For servers with up to two CPU cores, the default value is `4`. For three or more CPU cores, the default value is `8`. Local cluster installations, created with `yb-ctl` and `yb-docker-ctl`, use a value of `2` for this flag. Clusters created with `yugabyted` use a default value of `1`.
```

File: docs/content/latest/reference/configuration/yb-tserver.md [hunk 4]
```diff
@@ -347,7 +347,7 @@ On a per-table basis, the [`CREATE TABLE ... WITH TABLETS = <num>`](../../../api
 
 The number of shards per YB-TServer for each YSQL table when a user table is created.
 
-Default: `8`
+Default: `-1` (server internally sets default value). For servers with two or less CPU cores, then the default value is `2`. For servers with three or four CPU cores, the default value is 4. Beyond 4 cores, the default value is `8`. Local cluster installations, created with `yb-ctl` and `yb-docker-ctl`, use a value of `2` for this flag. Clusters created with `yugabyted` use a default value of 1.
```
Comment (polarweasel) [inline docs/content/latest/reference/configuration/yb-tserver.md:350]:
```suggestion
Default: `-1` (server internally sets default value). For servers with up to two CPU cores, the default value is `2`. For servers with three or four CPU cores, the default value is `4`. Beyond four cores, the default value is `8`. Local cluster installations, created with `yb-ctl` and `yb-docker-ctl`, use a value of `2` for this flag. Clusters created with `yugabyted` use a default value of `1`.
```

**Review by polarweasel — CHANGES_REQUESTED** [review]

File: docs/content/latest/reference/configuration/yb-master.md (hunk 2 above)
Comment (sanketkedia) [reply docs/content/latest/reference/configuration/yb-master.md:458]: Yes, it is intentional was based on a slack discussion involving Kannan and Karthik.

Comment (polarweasel) [reply docs/content/latest/reference/configuration/yb-master.md]: Great. Just double-checking. Thanks!

**Review by polarweasel — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**